Sources/XCTest/Public/XCTestCase.swift

```diff
@@ -99,7 +97,10 @@ open class XCTestCase: XCTest {
         XCTCurrentTestCase = self
         testRun.start()
         invokeTest()
-        failIfExpectationsNotWaitedFor(_allExpectations)
+
+        let allExpectations = XCTWaiter.subsystemQueue.sync { _allExpectations }
```

Comment:
This is a drive-by fix to an apparently un-synchronized read of the `_allExpectations` property